beps/0005-definitive-plugin-listing/README.md

```diff
@@ -0,0 +1,240 @@
+change are understandable. This may include API specs or even code snippets.
+If there's any ambiguity about HOW your proposal will be implemented, this is the place to discuss them.
+-->
+
+### Glossary
+
+### New Type `BackendFeatureMeta`
+
+```ts
+type BackendFeatureMeta =
+  | {
+      type: 'plugin';
+      pluginId: string;
+    }
+  | {
+      type: 'module';
+      pluginId: string;
+      moduleId: string;
+    };
+```
+
+### New `InstanceMetadataService`
+
+While we could attach the existing information to the `PluginMetadataService`, we propose a new service that handles instance-level information. The existing `PluginMetadataService` should reveal information about the plugin itself, its `pluginId`, dependencies or similar. The new `InstanceMetadataService` should give you information about the entire Backstage instance that you're interrogating. One could imagine this service also having information about instance URLs, health or gateway status.
+
+```ts
+interface InstanceMetadataService {
+  getInstalledFeatures: () => BackendFeatureMeta[];
+}
+```
+
+Currently, this would need to be bootstrapped directly in `BackendInitializer` or `BackstageBackend`. It may make sense to create a new `FeatureRegistry` class that can expose this instead similar to the `PluginMetadataService`.
+
+### New HTTP API to expose `InstanceMetadataService`
+
+We also propose exposing the new services through an HTTP API, so that these can be queried.
+
+```yaml
+paths:
+  /.backstage/instanceInfo/features/installed:
+    get:
+      summary: Get a list of installed features for this instance.
+      operationId: GetInstalledFeaturesByInstance
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  oneOf:
+                    - type: object
+                      properties:
+                        type:
+                          type: string
+                          enum:
+                            - plugin
+                        pluginId: { type: string }
+                    - type: object
+                      properties:
+                        type:
+                          type: string
+                          enum:
+                            - module
+                        pluginId: { type: string }
+                        moduleId: { type: string }
+```
+
+### New `DeploymentMetadataPlugin`
+
+This will be a new core plugin that will be provided through a new package. The idea is that it will show all information across your entire deployment. One can imagine this hosting aggregate information around instances in the deployment, like their internal and external URLs.
+
+```ts
+interface BackstageInstance {
+  url: {
+    internal: string;
+    external: string;
+  };
+}
+
+interface DeploymentMetadataPluginExtensionPoint {
```

Comment:
I think we'll want to provide these either with a custom implementation, or via configuration tbh, rather than mutating the service.

Reply:
updated in [05d1f2b](https://github.com/backstage/backstage/pull/25431/commits/05d1f2b5902124dcf7a087b4e3abf8e29fb8eb27)